src/tools/singlejar/output_jar.cc

```diff
@@ -192,12 +193,19 @@ int OutputJar::Doit(Options *options) {
 
   for (auto &rdesc : options_->resources) {
     // A resource description is either NAME or PATH:NAME
-    std::size_t colon = rdesc.find_first_of(':');
+    // Find the last ':' instead of the first because Windows uses ':' as volume
+    // separator in absolute path.
+    std::size_t colon = rdesc.find_last_of(':');
     if (0 == colon) {
       diag_errx(1, "%s:%d: Bad resource description %s", __FILE__, __LINE__,
                 rdesc.c_str());
     }
-    if (std::string::npos == colon) {
+    bool shouldNotSplit = colon == std::string::npos;
+#ifdef _WIN32
+    // If colon points to volume separator, don't split.
+    shouldNotSplit |= colon == 1 && blaze_util::IsAbsolute(rdesc);
```

Comment:
Yes, I meant `||=`, believing C++ had that operator. But that would've introduced a bug (if the operator existed), because `||` is short-circuit.

This indicates that (a) your code is correct because it ensures the right side is fully evaluated, but (b) it's very easy to break this code if one isn't aware of the subtle details.

Could you please rewrite it in more explicit form? (And rename `shouldNotSplit` to `shouldSplit`.)

```
if (colon == 1 && blaze_util::IsAbsolute(rdesc)) {
  shouldSplit = false;
}
```

Reply:
Done.